netlify/functions/webhooks.js

```diff
@@ -0,0 +1,77 @@
+import { App, Octokit } from 'octokit';
+import githubApp from '../../github-app';
+import pino from 'pino';
+
+
+const log = pino();
+const octokitLog = log.child({ name: 'octokit' });
+
+const setupApp = async () => {
+  try {
+    const app = new App({
+      appId: process.env.GITHUB_APP_ID,
+      privateKey: process.env.GITHUB_APP_PRIVATE_KEY,
+      oauth: {
+        clientId: process.env.GITHUB_OAUTH_CLIENT_ID,
+        clientSecret: process.env.GITHUB_OAUTH_CLIENT_SECRET
+      },
+      webhooks: {
+        secret: process.env.GITHUB_WEBHOOK_SECRET,
+      },
+      Octokit: Octokit.defaults({
+        userAgent: 'gr2m/github-app-slack-demo',
+      }),
+      log: {
+        debug: octokitLog.debug.bind(octokitLog),
+        info: octokitLog.info.bind(octokitLog),
+        warn: octokitLog.warn.bind(octokitLog),
+        error: octokitLog.error.bind(octokitLog),
+      },
+    });
+
+    const { data: appInfo } = await app.octokit.request('GET /app');
+    log.info({ slug: appInfo.slug, url: appInfo.html_url }, `Authenticated`);
+
+    await githubApp(app);
+    
+    return app
+  } catch (error) {
+    log.error(error, 'Failed to set up app');
+    throw error;
+  }
+};
+
+/**
+ * Netlify function to handle webhook event requests from GitHub
+ *
+ * @param {import("@netlify/functions").HandlerEvent} event
+ * @param {import("@netlify/functions").HandlerContext} context
+ */
+export const handler = async (event, context) => {
+  if (event.httpMethod !== "POST"){
+    return {
+      statusCode: 405,
+      body: JSON.stringify({ error: 'Method not allowed' }),
+    }
+  }
+  
+  try {
+    const app = await setupApp();
+    await app.webhooks.verifyAndReceive({
+      id: event.headers["X-GitHub-Delivery"] ||
+      event.headers["x-github-delivery"],
+      name: event.headers["X-GitHub-Event"] || event.headers["x-github-event"],
+      signature: event.headers["X-Hub-Signature-256"] ||
+      event.headers["x-hub-signature-256"],
+      payload: JSON.parse(event.body)
+    })
+  } catch (error) {
+    log.error(error, 'Handler error');
+    
+  }finally{
+    return {
+      statusCode: 200,
+      body: JSON.stringify({ ok: true }),
+    }
+  }
```

Comment:
return different codes depending on whether an error occurred or not.

We will also need to account for webhook timeouts. GitHub is only waiting for 10s before it times out the request.

We can copy the implementation from here to account for that:
https://github.com/octokit/webhooks.js/blob/5b4db42af275a7a83a5041d662fc91cd932201a2/src/middleware/node/middleware.ts#L86-L111

We could probably make some thing more elegant using [Promise.race()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/race)

I'm also okay to leave this out of scope for this pull request and to create a follow up issue to be addressed later